addons/mail/models/mail_message.py

```diff
@@ -417,6 +411,10 @@ def message_format(self):
         ])
         message_tree = dict((m.id, m) for m in self.sudo())
         self._message_read_dict_postprocess(message_values, message_tree)
+        active_ids = self.env['mail.notification'].sudo().search([
```

Comment:
instead of search records on mail.notification model, can possible to fetch only active notifications on read 'needaction_partner_ids'  field ?